### DISCUSSION TITLE
Inject-aware context documentation changes

### DISCUSSION THREAD

**Discussion by niksy** [vuejs/pinia#2556] [Ideas]
I’ve noticed some parts in documentation regarding SSR that I think could benefit from changes to make them more understandable:

* In part [on using outside `setup()`](https://pinia.vuejs.org/ssr/#Using-the-store-outside-of-setup-), there is example where it says you need to pass Pinia instance to `useStore` call inside vue-router guard, but that is not necessary [as of vue-router 4.2](https://github.com/vuejs/router/commit/909c45be199fbf48e55f3f7e681e93c259e82a14) since guards are inject-aware and proper Pinia instance would be injected. Is that correct? Should that be changed to use some other example?
* In part [on using outside components](https://pinia.vuejs.org/core-concepts/outside-component-usage.html#SSR-Apps), it says "When dealing with Server Side Rendering, you will have to pass the pinia instance to useStore()." I think passing instance is not necessary if you call `useStore` inside `app.runWithContext`, is that true? Can this section be expanded to include example of that approach?

I’m willing to submit changes to documentation if you think these statements are correct.

**Comment by posva** [comment]
1. Yes, it should be updated with a tip that explains this is no longer necessary
2. I'm not sure if `app.runWithContext()` is intended for app-level usage. I added it for libraries to benefit from the injection context. I think that section of the docs and the linked ssr page is fine

**Comment by niksy** [comment]
> I'm not sure if app.runWithContext() is intended for app-level usage. I added it for libraries to benefit from the injection context. I think that section of the docs and the linked ssr page is fine

But is it correct that it can be used for app-level needs to avoid passing Pinia instance and instead get it from app injection context, even in SSR applications? There is no mention in [official documentation](https://vuejs.org/api/application.html#app-runwithcontext) regarding intended usage.
